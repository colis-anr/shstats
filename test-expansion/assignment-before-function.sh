#!/bin/dash

f() {
    echo coocoo
}
x=old
x=new f
# we don't know statically the value of $x
echo $x
