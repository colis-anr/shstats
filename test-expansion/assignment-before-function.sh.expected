[
  [
    "CompleteCommand_CList",
    [
      "CList_AndOr",
      [
        "AndOr_Pipeline",
        [
          "Pipeline_PipeSequence",
          [
            "PipeSequence_Command",
            [
              "Command_FunctionDefinition",
              [
                "FunctionDefinition_Fname_Lparen_Rparen_LineBreak_FunctionBody",
                [ "Fname_Name", [ "Name", "f" ] ],
                [ "LineBreak_Empty" ],
                [
                  "FunctionBody_CompoundCommand",
                  [
                    "CompoundCommand_BraceGroup",
                    [
                      "BraceGroup_LBrace_CompoundList_RBrace",
                      [
                        "CompoundList_NewLineList_Term_Separator",
                        [ "NewLineList_NewLine" ],
                        [
                          "Term_AndOr",
                          [
                            "AndOr_Pipeline",
                            [
                              "Pipeline_PipeSequence",
                              [
                                "PipeSequence_Command",
                                [
                                  "Command_SimpleCommand",
                                  [
                                    "SimpleCommand_CmdName_CmdSuffix",
                                    [
                                      "CmdName_Word",
                                      [
                                        "Word",
                                        "echo",
                                        [ [ "WordName", "echo" ] ]
                                      ]
                                    ],
                                    [
                                      "CmdSuffix_Word",
                                      [ "Word", "coocoo", [] ]
                                    ]
                                  ]
                                ]
                              ]
                            ]
                          ]
                        ],
                        [
                          "Separator_NewLineList",
                          [ "NewLineList_NewLine" ]
                        ]
                      ]
                    ]
                  ]
                ]
              ]
            ]
          ]
        ]
      ]
    ]
  ],
  [
    "CompleteCommand_CList",
    [
      "CList_AndOr",
      [
        "AndOr_Pipeline",
        [
          "Pipeline_PipeSequence",
          [
            "PipeSequence_Command",
            [
              "Command_SimpleCommand",
              [
                "SimpleCommand_CmdPrefix",
                [
                  "CmdPrefix_AssignmentWord",
                  [ [ "Name", "x" ], [ "Word", "old", [] ] ]
                ]
              ]
            ]
          ]
        ]
      ]
    ]
  ],
  [
    "CompleteCommand_CList",
    [
      "CList_AndOr",
      [
        "AndOr_Pipeline",
        [
          "Pipeline_PipeSequence",
          [
            "PipeSequence_Command",
            [
              "Command_SimpleCommand",
              [
                "SimpleCommand_CmdPrefix_CmdWord",
                [
                  "CmdPrefix_AssignmentWord",
                  [ [ "Name", "x" ], [ "Word", "new", [] ] ]
                ],
                [ "CmdWord_Word", [ "Word", "f", [] ] ]
              ]
            ]
          ]
        ]
      ]
    ]
  ],
  [
    "CompleteCommand_CList",
    [
      "CList_AndOr",
      [
        "AndOr_Pipeline",
        [
          "Pipeline_PipeSequence",
          [
            "PipeSequence_Command",
            [
              "Command_SimpleCommand",
              [
                "SimpleCommand_CmdName_CmdSuffix",
                [
                  "CmdName_Word",
                  [ "Word", "echo", [ [ "WordName", "echo" ] ] ]
                ],
                [
                  "CmdSuffix_Word",
                  [ "Word", "$x", [ [ "WordLiteral", "$x" ] ] ]
                ]
              ]
            ]
          ]
        ]
      ]
    ]
  ]
]